commit e855558f84a4fa1df8790ef6a5b5a98294ffc24f
Author: phanirithvij <phanirithvij2000@gmail.com>
Date:   Sun May 25 21:33:50 2025 +0530

    Revert "nixos/limine: don't modify boot order on bootloader update"
    
    This reverts commit 95030d7eb31bcba8b01a0d05673e4a508bdc7017.

diff --git a/nixos/modules/system/boot/loader/limine/limine-install.py b/nixos/modules/system/boot/loader/limine/limine-install.py
index 3407d778d62b..94d47a3169c1 100644
--- a/nixos/modules/system/boot/loader/limine/limine-install.py
+++ b/nixos/modules/system/boot/loader/limine/limine-install.py
@@ -388,16 +388,9 @@ def main():
                 efibootmgr = os.path.join(config('efiBootMgrPath'), 'bin', 'efibootmgr')
                 efi_partition = find_mounted_device(config('efiMountPoint'))
                 efi_disk = find_disk_device(efi_partition)
-
-                efibootmgr_output = subprocess.check_output([efibootmgr], stderr=subprocess.STDOUT, universal_newlines=True)
-                create_flag = '-c'
-                # Check the output of `efibootmgr` to find if limine is already installed and present in the boot record
-                if matches := re.findall(r'Boot[0-9a-fA-F]{4}\*? Limine', efibootmgr_output):
-                    create_flag = '-C' # if present, keep the same boot order
-
                 efibootmgr_output = subprocess.check_output([
                     efibootmgr,
-                    create_flag,
+                    '-c',
                     '-d', efi_disk,
                     '-p', efi_partition.removeprefix(efi_disk).removeprefix('p'),
                     '-l', f'\\efi\\limine\\{boot_file}',

